src/Storages/MySQL/MySQLSettings.h

```diff
@@ -17,6 +17,7 @@ class ASTStorage;
 #define LIST_OF_MYSQL_SETTINGS(M) \
     M(UInt64, connection_pool_size, 16, "Size of connection pool (if all connections are in use, the query will wait until some connection will be freed).", 0) \
     M(UInt64, connection_max_tries, 3, "Number of retries for pool with failover", 0) \
+    M(Bool, connection_no_block, false, "Do not wait for connection if all connections are in use.", 0) \
```

Comment:
Better to add setting connection_timeout, and if timeout is 0 it is like connection_no_block=true.

Reply:
Yes, this is a good idea (I had the same in mind, however I went with connection_no_block since I want to test it ASAP), I will update the patch.